### PR TITLE
[Experimental] Add e2e tests for the new stock status filter block

### LINKED ISSUE #42252: E2E tests: Stock Filter
Convert tests to Playwright edition. Remove legacy test after conversion. Fix any flaky tests.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/42252 by adding basic e2e tests for the stock status filter.

In future when we add a dynamic templating solution we can more easily test a bunch of different options being passed to the block.

### How to test the changes in this Pull Request:

1. See the CI pass

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			page,
+		} ) => {
+			await page.goto( '/product-filters-stock-status-block/' );
+
+			const stockStatusCheckboxes = page.locator(
+				'.wc-block-components-checkbox__input'
+			);
+
+			// Out of stock
+			stockStatusCheckboxes.nth( 1 ).check();
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts:33]: What do you think about using `getByLabel()` here? It makes the code easier to read and it should be more resilient as we don't depend on a specific CSS selector.

```suggestion
			const outOfStockCheckbox = page.getByLabel( 'Out of stock' );
			outOfStockCheckbox.click();
```

**Review by Aljullu — APPROVED** [review]
LGTM! I left one minor suggestion, but pre-approving.

File: plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts (hunk 1 above)
Comment (samueljseay) [reply plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts:33]: Great idea thanks will fix :)

Comment (samueljseay) [reply plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts:33]: I've decided to migrate these tests to https://github.com/woocommerce/woocommerce/pull/44223 which is dynamically creating post content. I've incorporated these suggested changes in that PR. 🙇

**Pull request closed without merging**